### ISSUE TITLE
The sauserprefs connection string

### ISSUE THREAD

**Issue by akissz** [johndoh/roundcube-sauserprefs#77]
The sauserprefs connection string should use the primary roundcube database connection string. There has been bugs about this as far back as the early 2000s. It seems this still has not been implemented or improved. I think using a separate connection string causes bugs.

**Comment by johndoh** [comment]
When you say "should use" please can you explain the issue that you are facing in more detail? Spamassassin and Roundcube are separate applications and so while nothing is stopping them from sharing a database it cannot be assumed that they do. With the current config options you are free to define the connection strings as required.

> There has been bugs about this as far back as the early 2000s.

I do not know what bugs you are referring too, this plugin was created in 2009.

May be I am missing something, it would be trivial to add a fallback so that if the sauserprefs db connection strings are set to `null` then it would use the Roundcube connection strings but I don't know how useful that would really be. Is the sort of thing you are looking for and if so please can you provide an explanation of the use case?

**Comment by akissz** [comment]
Yes I think the fallback idea you mentioned would be sufficient, good idea! Then there would be less errors. Or also have a tab in settings to allow for entry for alternative connection, this way seems more common. I have only used roundcube a short while and I get the feeling the community is stuck in this way of thinking, separate applications as you put it. And having unnecessary things like extra files in every single directory which is madness to me. Separate applications is great but that also ends up making bad situations for backwards compatibility and user interface bugs. I'm saying that when I googled this database issue it shows up in google very far back and it made me wonder why there are 2 database connections because I have never seen this in the hundreds of other programs that I use, there normally is only 1 connection. Maybe it is fine that little program in roundcube is better with 2 connections. :) Hope my comments help with ideas and are not a bother.

**Comment by johndoh** [comment]
> I'm saying that when I googled this database issue it shows up in google very far back and it made me wonder why there are 2 database connections because I have never seen this in the hundreds of other programs that I use, there normally is only 1 connection.

Roundcube, a web based IMAP client, and Spamassassin, a server side spam filtering tool, are two completely separate, unrelated applications. There is no reason for one to even know the other exists. If someone chooses to install this plugin into their Roundcube setup then I assume they already have a working Spamassassin installation and then, yes, the plugin needs to be configured to connect to that existing installation and that requires a config file.
